src/app/pages/MovieDetails/MovieDetails.tsx

```diff
@@ -8,11 +8,25 @@ import styles from './MovieDetails.module.css';
 import Rating from '../../components/Rating/Rating';
 import NavigationGenre from '../../components/NavigationGenre/NavigationGenre';
 import NavBar from '../../components/NavBar/NavBar';
+import { Movie } from '../../../types';
 
-const categoriesArray = ['Movie', 'Adventure', 'Comedy', 'Family'];
+// const categoriesArray = ['Movie', 'Adventure', 'Comedy', 'Family'];
 
 function MovieDetails(): JSX.Element {
-  const { name } = useParams<{ name: string }>();
+  const { id } = useParams<{ id: string }>();
+  const [movie, setMovie] = useState<Movie>();
+
+  useEffect(() => {
+    fetch(`/api/movies/${id}`)
+      .then((response) => response.json())
+      .then((movie) => setMovie(movie));
+  }, []);
+
+  if (!movie) {
+    return <div>Loading...</div>;
+  }
+
+  const genre = movie.genres.map((genres) => genres.name);
```

Comment:
```suggestion
  const genre = movie.genres.map((genre) => genre.name);
```
Maybe change it to singular naming, since it is one object.

Reply:
```suggestion
  const genreNames = movie.genres.map((genre) => genre.name);
```